src/Microsoft.ML.Data/Transforms/TrainAndScoreTransform.cs

```diff
@@ -131,7 +152,36 @@ public sealed class Arguments : ArgumentsBase<SignatureTrainer>
 
         internal const string Summary = "Trains a predictor, or loads it from a file, and runs it on the data.";
 
+        /// <summary>
+        /// Convenience method for creating <see cref="TrainAndScoreTransform"/>.
+        /// The <see cref="TrainAndScoreTransform"/> allows for model stacking (i.e. to combine information from multiple predictive models to generate a new model)
+        /// in the pipeline by training a model first and then using the scores from the trained model.
+        ///
+        /// Unlike <see cref="ScoreTransform"/>, the <see cref="TrainAndScoreTransform"/> trains the model on the fly as name indicates.
+        /// </summary>
+        /// <param name="env">Host Environment.</param>
+        /// <param name="input">Input <see cref="IDataView"/>.</param>
+        /// <param name="trainer">The <see cref="ITrainer"/> object i.e. the learning algorithm that will be used for training the model.</param>
+        /// <param name="featureColumn">Role name for features.</param>
+        /// <param name="labelColumn">Role name for label.</param>
+        /// <returns></returns>
+        public static IDataTransform Create(IHostEnvironment env, IDataView input, ITrainer trainer, string featureColumn = DefaultColumnNames.Features, string labelColumn = DefaultColumnNames.Label)
+        {
+            var args = new Arguments()
+            {
+                FeatureColumn = featureColumn,
+                LabelColumn = labelColumn
+            };
+
+            return Create(env, args, trainer, input);
+        }
+
         public static IDataTransform Create(IHostEnvironment env, Arguments args, IDataView input)
+        {
+            return Create(env, args, args.Trainer.CreateInstance(env), input);
+        }
+
+        private static IDataTransform Create(IHostEnvironment env, Arguments args, ITrainer trainer, IDataView input)
```

Comment:
Actually, a more common pattern is to use AssertValue in private methods, and CheckValue in public methods. 

The purpose of the Check* is twofold: first, we want to enforce the calling contract, and second, we want to actually EXPLICITLY SPECIFY the calling contract.
So that you can look at any argument and immediately, within 3 lines of code, tell whether it can be null or not etc.

---
In reply to: [206362456](https://github.com/dotnet/machinelearning/pull/614#discussion_r206362456) [](ancestors = 206362456)

Reply:
Moved all the checks in the public methods instead. 

---
In reply to: [206367797](https://github.com/dotnet/machinelearning/pull/614#discussion_r206367797) [](ancestors = 206367797,206362456)